test/python/transpiler/test_elide_permutations.py

```diff
@@ -0,0 +1,425 @@
+        self.assertEqual(res, expected)
+
+    def test_permutation_at_beginning(self):
+        """Test permutation in beginning of bell is elided."""
+        qc = QuantumCircuit(3, 3)
+        qc.append(PermutationGate([2, 1, 0]), [0, 1, 2])
+        qc.h(0)
+        qc.cx(1, 2)
+        qc.barrier(0, 1, 2)
+        qc.measure(0, 0)
+        qc.measure(1, 1)
+        qc.measure(2, 2)
+
+        expected = QuantumCircuit(3, 3)
+        expected.h(2)
+        expected.cx(1, 0)
+        expected.barrier(0, 1, 2)
+        expected.measure(2, 0)
+        expected.measure(1, 1)
+        expected.measure(0, 2)
+
+        res = self.swap_pass(qc)
+        self.assertEqual(res, expected)
+
+    def test_permutation_at_end(self):
+        """Test permutation at end of bell is elided."""
+        qc = QuantumCircuit(3, 3)
+        qc.h(0)
+        qc.cx(1, 2)
+        qc.barrier(0, 1, 2)
+        qc.measure(0, 0)
+        qc.measure(1, 1)
+        qc.measure(2, 2)
+        qc.append(PermutationGate([2, 1, 0]), [0, 1, 2])
+
+        expected = QuantumCircuit(3, 3)
+        expected.h(0)
+        expected.cx(1, 2)
+        expected.barrier(0, 1, 2)
+        expected.measure(0, 0)
+        expected.measure(1, 1)
+        expected.measure(2, 2)
+
+        res = self.swap_pass(qc)
+        self.assertEqual(res, expected)
+
+    def test_swap_and_permutation(self):
+        """Test a combination of swap and permutation gates."""
+        qc = QuantumCircuit(3, 3)
+        qc.append(PermutationGate([2, 1, 0]), [0, 1, 2])
+        qc.swap(0, 2)
+        qc.h(0)
+        qc.cx(1, 2)
+        qc.barrier(0, 1, 2)
+        qc.measure(0, 0)
+        qc.measure(1, 1)
+        qc.measure(2, 2)
+        expected = QuantumCircuit(3, 3)
+        expected.h(0)
+        expected.cx(1, 2)
+        expected.barrier(0, 1, 2)
+        expected.measure(0, 0)
+        expected.measure(1, 1)
+        expected.measure(2, 2)
+
+        res = self.swap_pass(qc)
+        self.assertEqual(res, expected)
+
+    def test_permutation_before_measure(self):
+        """Test permutation before measure is elided."""
+        qc = QuantumCircuit(3, 3)
+        qc.h(0)
+        qc.cx(1, 2)
+        qc.barrier(0, 1, 2)
+        qc.append(PermutationGate([1, 2, 0]), [0, 1, 2])
+        qc.measure(0, 0)
+        qc.measure(1, 1)
+        qc.measure(2, 2)
+
+        expected = QuantumCircuit(3, 3)
+        expected.h(0)
+        expected.cx(1, 2)
+        expected.barrier(0, 1, 2)
+        expected.measure(1, 0)
+        expected.measure(2, 1)
+        expected.measure(0, 2)
+
+        res = self.swap_pass(qc)
+        self.assertEqual(res, expected)
+
+
+class TestElidePermutationsInTranspileFlow(QiskitTestCase):
+    """
+    Test elide permutations in the full transpile pipeline, especially that
+    "layout" and "final_layout" attributes are updated correctly
+    as to preserve unitary equivalence.
+    """
+
+    # ToDo: update the tests when ElidePermutations is merged into transpiler flow.
```

Comment:
What is meant by this? To add preset passmanager tests?

Reply:
Yeah, I assume @alexanderivrii meant that after #12111 merged we can use the preset pass managers for testing here. Although TBH, personally I'm fine leaving the workflow tests explicit with smaller pass managers like this.

Reply:
I dropped the comment as part of: https://github.com/Qiskit/qiskit/pull/9523/commits/f5b6778e524ff5af540cf66fb0a42eb0a2a0b6c8